draft/0005-impersonation.md

```diff
@@ -0,0 +1,260 @@
+  "iat": 1516239022,
+  "act": {
+    "sub": "456"
+  }
+}
+```
+
+## Subject token
+
+The `subject_token` parameter in the token exchange request represents the identity of the party on behalf of whom the request is being made. The `subject_token` is a special one-time access token that is issued to the authorized user by Logto's management API. The `subject_token` is an opaque token that is not intended to be used directly by the client application. Instead, the client application exchanges the `subject_token` for a new access token that represents the identity of the impersonated user. Because the `subject_token` is only consumed by Logto as the authorization server, it does not need to be a JWT.
+
+### Issuing a subject token
+
+#### Request
+
+The client makes a request to the tenant's management API to issue a subject token. The following parameters are included in the HTTP request entity-body using the `application/json` format.
+
+1. `userId`: REQUIRED. The ID of the user on behalf of whom the request is being made.
+2. `expiresIn`: OPTIONAL. The lifetime in seconds of the subject token. The default value is 600 seconds.
+
+#### Response
+
+If the request to issue a subject token is successful, the tenant's management API returns the subject token in the response.
+
+1. `subject_token`: REQUIRED. The subject token that represents the identity of the user.
+2. `expires_in`: REQUIRED. The lifetime in seconds of the subject token.
+
+#### Example subject token issuance
+
+The following example demonstrates a request to issue a subject token and the response. This request is made by a machine-to-machine application that has the necessary permissions to issue subject tokens.
+
+```
+POST /api/subject-tokens HTTP/1.1
+Host: example.logto.app
+Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9...
+Content-Type: application/json
+
+{
+  "userId":"123456",
+  "expiresIn":600
+}
+```
+
+Logto validates the `userId` and permissions to issue a subject token. If the request is valid, Logto generates a new subject token that represents the identity of the user and returns it in the response.
+
+```
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "subject_token":"accVkjcJyb4BWCxGsndESCJQbdFMogUC5PbRDqceLTC",
+  "expires_in":600
+}
+```
+
+## Additional claims in the access token
+
+### `act`
+
+When a user impersonates another user, the access token issued by Logto contains an additional `act` claim that represents the identity of the impersonated user. The `act` claim is an object that contains the following fields.
+
+1. `sub`: REQUIRED. The identity of the acting user. Typically, this will be the party that is authorized to use the requested security token and act on behalf of the subject.
+
+This is an example JWT access token with the `act` claim.
+
+```json
+{
+  "aud": "https://resource.example.com",
+  "iss": "https://example.logto.app",
+  "exp": 1443904177,
+  "sub": "user2",
+  "act": {
+    "sub": "user1"
```

Comment:
1. if this is required, where does it come from? i assume there should be a field to fill when requesting Management API for the subject token
2. can the sub be an app?

Reply:
Add a param `actorId` to the request

Reply:
didn't see it though

Reply:
it's ok i saw the update